pkg/controller/container-runtime-config/helpers.go

```diff
@@ -495,29 +496,64 @@ func validateUserContainerRuntimeConfig(cfg *mcfgv1.ContainerRuntimeConfig) erro
 // getValidBlockedRegistries gets the blocked registries in the image spec and validates that the user is not adding
 // the registry being used by the payload to the list of blocked registries.
 // If the user is, we drop that registry and continue with syncing the registries.conf with the other registry options
-func getValidBlockedRegistries(releaseImage string, imgSpec *apicfgv1.ImageSpec) ([]string, error) {
+func getValidBlockedRegistries(releaseImage string, imgSpec *apicfgv1.ImageSpec, icspRules []*apioperatorsv1alpha1.ImageContentSourcePolicy) ([]string, error) {
 	if imgSpec == nil {
 		return nil, nil
 	}
 
-	var blockedRegs []string
+	var (
+		blockedRegs []string
+		blockErr    []string
+	)
 
 	// Get the registry being used by the payload from the releaseImage
 	ref, err := reference.ParseNamed(releaseImage)
 	if err != nil {
 		return nil, errParsingReference
 	}
-	payloadReg := reference.Domain(ref)
-	for i, reg := range imgSpec.RegistrySources.BlockedRegistries {
+	payloadRepo := reference.TrimNamed(ref).String()
```

Comment:
(I think this can be just `ref.Name()`. This works fine.)

Reply:
fixed